common/achievement/create.ts

```diff
@@ -0,0 +1,112 @@
+import { Prisma, achievement_type_enum } from '@prisma/client';
+import { Achievement_template, achievement_template_for_enum } from '../../graphql/generated';
+import { ActionID, SpecificNotificationContext } from '../notification/actions';
+import { prisma } from '../prisma';
+import { TemplateSelectEnum, getAchievementTemplates } from './template';
+import tracer from '../logger/tracing';
+import { AchievementToCheck } from './types';
+
+async function findUserAchievement<ID extends ActionID>(
+    templateId: number,
+    userId: string,
+    context: SpecificNotificationContext<ID>
+): Promise<AchievementToCheck> {
+    const keys = context ? Object.keys(context) : [];
+    const userAchievement = await prisma.user_achievement.findFirst({
+        where: {
+            templateId,
+            userId,
+            AND: keys.map((key) => {
```

Comment:
Do we actually need to search for the whole context here? Shouldn't it be enough to search by templateID + relation?